### PR TITLE
S2 transformers list

### PR DESCRIPTION
**Pull request by BelindaWilliams**
S2->Main (for trainer review)

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+?>
+
+<!DOCTYPE html>
+
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <title>Transformers</title>
```
Comment (nealioa) [inline index.php]: Please move the `title` further down so that all the `meta` tags are grouped together.

File: index.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                <div class="filter-select col-lg-2 col-sm-12 form-check form-switch d-flex justify-content-center">
+                    <input type="checkbox" class="form-check-input" id="flexSwitchCheckChecked" checked></input>
+                    <label class="form-check-label" for="flexSwitchCheckChecked">Insecticons</label>
+                </div>
+
+                <div class="filter-select col-lg-2 col-sm-12 form-check form-switch d-flex justify-content-center">
+                    <input type="checkbox" class="form-check-input" id="flexSwitchCheckChecked" checked></input>
+                    <label class="form-check-label" for="flexSwitchCheckChecked">Decepticons</label>
+                </div>
+                <div class="col-lg-1 col-sm-0"></div>
```
Comment (nealioa) [inline index.php:56]: For consistency, there should be a blank line before this one.

File: src/Abstracts/Transformer.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Transformers\Abstracts;
+
+class Transformer
```
Comment (nealioa) [inline src/Abstracts/Transformer.php]: Why is this class in a folder called 'Abstracts' when it is not actually an abstract class?
Maybe the keyword `abstract` is missing?

File: src/Models/IndexModel.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Transformers\Models;
+
+class IndexModel
+{
+    private static ?IndexModel $instance = null;
+
+    public array $transformerList = [];
```
Comment (nealioa) [inline src/Models/IndexModel.php]: Why is this property public?

File: src/Models/IndexModel.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    public static function getInstance(): IndexModel
+    {
+        if (!self::$instance) {
+            self::$instance = new IndexModel();
+        }
+        return self::$instance;
+    }
+
+    public function setTransformers($transformers)
```
Comment (nealioa) [inline src/Models/IndexModel.php]: Related to above comment.
The property is public. So no need for a setter.
Or should the property be private and a getter is also required?

File: src/ViewHelpers/IndexViewHelper.php [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        return $str;
+    }
+
+    /**
+     * Creates a html string with all Transformers cards
+     * @return string
+     */
+    public function createTransformersList(): string
+    {
+        $str = "";
```
Comment (nealioa) [inline src/ViewHelpers/IndexViewHelper.php]: Single quotes have been used in above method.

**Review by nealioa — CHANGES_REQUESTED** [review]

File: index.php (hunk 1 above)
Comment (henridvf) [reply index.php]: Corrected

File: src/Models/IndexModel.php (hunk 4 above)
Comment (henridvf) [reply src/Models/IndexModel.php]: Changed to private array

File: src/Models/IndexModel.php (hunk 5 above)
Comment (henridvf) [reply src/Models/IndexModel.php]: Yes, changed the array itself to private and now provided getter and setter

File: src/ViewHelpers/IndexViewHelper.php (hunk 6 above)
Comment (henridvf) [reply src/ViewHelpers/IndexViewHelper.php]: Corrected to single quotes

File: src/Abstracts/Transformer.php (hunk 3 above)
Comment (henridvf) [reply src/Abstracts/Transformer.php]: Is indeed not an abstract. Changed folder to 'Entities'.

**Review by nealioa — APPROVED** [review]

**Pull request merged**